### ISSUE TITLE
Removing definitions seems to not work

### ISSUE THREAD

**Issue by matentzn** [ontodev/robot#532]
I am trying to run:

```
robot remove --input o.owl --term-file simple-rm.txt --axioms annotation --trim false -o $@.owl
```

In order to remove all axioms of this kind:
```
<owl:Axiom>
        <owl:annotatedSource rdf:resource="http://purl.obolibrary.org/obo/FBcv_0000002"/>
        <owl:annotatedProperty rdf:resource="http://purl.obolibrary.org/obo/IAO_0000115"/>
        <owl:annotatedTarget rdf:datatype="http://www.w3.org/2001/XMLSchema#string">Phenotype that is a defect in thermotaxis (GO:0043052). &apos;thermotaxis&apos; is defined as: &apos;$sub_GO:0043052&apos;</owl:annotatedTarget>
        <oboInOwl:hasDbXref rdf:datatype="http://www.w3.org/2001/XMLSchema#string">FBC:DOS</oboInOwl:hasDbXref>
</owl:Axiom>
```

I tried to add hasDBXref iri to the term file as well, and tried signature true. The goal is to remove all definitions of the classes in the term file, with or without axiom annotations on them.. What is the best way to accomplish this?

```
git clone https://github.com/monarch-ebi-dev/robot_test.git
cd robot_test
make fail_remove_definitions
diff o.owl fail_remove_definitions.owl
```

Thaaaanks! :)

**Comment by rctauber** [comment]
To remove all definition axioms:
```
robot remove --input o.owl \
 --term IAO:0000115 --trim true --output o2.owl
```

You want to include `--trim true` so that *any* axiom containing `IAO:0000115` is removed. `IAO:0000115` is the only term you need to include to remove, and it will automatically handle the axioms with annotations on them.

**Comment by matentzn** [comment]
Thanksgiving Becky! Thing is, i want to delete only those definitions for the terms in the term file. Is that possible?

**Comment by rctauber** [comment]
Hmm. I'm not sure if that's possible with the way `remove` is currently set up. When using `--trim false`, you would also need to include the object of the axiom, i.e. the string literal definition.

A SPARQL update might be a better option in this case. Another option is to discuss adding another selector that returns the objects of any axioms that include both the subject AND predicate in the term set. This might be a little messy though...

**Comment by matentzn** [comment]
The literal needs to be included as well? hmmmm.. but should the `--signature true` have taken car of this? My intuition was that --signature true makes sure that, whatever we do, we only compare the signature, and not the set of OWLObjects, or am I mistaken here?

**Comment by rctauber** [comment]
I've been working on reworking the methods related to the `--signature` option. Could you please test your examples against this new branch? https://github.com/rctauber/robot/tree/532-fix

**Comment by matentzn** [comment]
Can you make a DO NOT MERGE pull request on the normal robot repo? I have actually never build robot :P will try that. We should also most definitely wait for James to comment on this issue.. Better to know wether we are consistent with the vision. Thanks Becky! :)

**Comment by rctauber** [comment]
If you `git clone` my repo, you should be able to checkout the `532-fix` branch and then build the ROBOT jar (`mvn clean package`) from there. Let me know if that doesn't work for you. Thanks!

**Comment by matentzn** [comment]
Ok. because we need to test ROBOT in the context of ODK builds, I wrote a script that builds a fresh ODK container with the branch build of ROBOT, see [here](https://github.com/monarch-ebi-dev/robot_test/blob/master/README.md). 

I currently have a bit of a processing backlog, but will come back to you once I get to actually run the tests with the thus modified odk :P Thanks.

**Comment by beckyjackson** [comment]
Hi @matentzn - it's been a bit of time since this issue was opened, sorry for not following up sooner. Are you still having this issue?

**Comment by matentzn** [comment]
I put this on my todo pile and will let you know soon. Thanks!

**Comment by matentzn** [comment]
Hey @beckyjackson is it easy for you to supply me with the correct final robot command for this and the right JAR file? I tried the latest development version of ROBOT and that does not work, but I guess this feature was never merged in?

**Comment by jamesaoverton** [comment]
The `532-fix` branch is on @beckyjackson's fork of this repo. I made a PR in this repo #585. But that didn't trigger Jenkins until I used the "Switch branches/tags" dropdown to "Create branch 532-fix" in this repo. Now it's available here: https://build.obolibrary.io/job/ontodev/job/robot/job/532-fix/

**Comment by matentzn** [comment]
Perfect thanks @jamesaoverton ; tested it. This command works:

```
./robot remove --input o.owl --term-file simple-rm.txt --axioms annotation --trim false --preserve-structure false --signature true -o o_no_defs.owl
```

However, I just want to note that the term file does need the signature of axiom annotations present as well. Lets say I want to remove all definitions `A IAO:0000115 "def 1"` specified in my term list, but some of them happen to have annotation assertions; i need to know beforehand exactly which annotation properties they have, for example in the case of `[A IAO:0000115 "def 1"] -[dbxref]-'flybase:1'`, i need to add dbxref to the term list. I am not entirely opposed to that for now, but I just wanted to mention it so you can can make a decision whether it would be good to have, for example, a parameter like --ignore axiom annotations. Up to you! Thanks for this feature anyways, my tests leave me satisfied :)

**Comment by beckyjackson** [comment]
I think having an `--ignore-axiom-annotations <true/false>` option would be great.

**Comment by beckyjackson** [comment]
I just made a new issue #600 to discuss the axiom annotations so we can keep track of that better. This definition fix is available in v1.5.0.
